test/DynamoCoreWpfTests/PackageManagerUITests.cs

```diff
@@ -1235,11 +1235,81 @@ public void PackageManagerConflictsUnloadedWithBltInPackage()
+        [Description("User tries to download packages might conflict with an existing package in error state")]
+        public void PackageManagerErrorStatePackages()
+        {
+            var pathMgr = ViewModel.Model.PathManager;
+            var pkgLoader = GetPackageLoader();
+
+            // Load a package
+            var packageLocation = Path.Combine(BuiltinPackagesTestDir, "SignedPackage2");
+            pkgLoader.ScanPackageDirectory(packageLocation);
+            var pkg = pkgLoader.LocalPackages.Where(x => x.Name == "SignedPackage").FirstOrDefault();
+            Assert.IsNotNull(pkg);
+
+            pkgLoader.LoadPackages(new List<Package>() { pkg });
+
+            pkg.LoadState.SetAsError("Test error");
+
+            // Simulate the user downloading the same package from PM
+            var mockGreg = new Mock<IGregClient>();
+            mockGreg.Setup(x => x.Execute(It.IsAny<PackageDownload>())).Throws(new Exception("Failed to get your package!"));
+
+            var client = new Dynamo.PackageManager.PackageManagerClient(mockGreg.Object, MockMaker.Empty<IPackageUploadBuilder>(), string.Empty);
+            var pmVm = new PackageManagerClientViewModel(ViewModel, client);
+
+            var dlgMock = new Mock<MessageBoxService.IMessageBox>();
+            dlgMock.Setup(m => m.Show(It.Is<string>(x => x.Contains("conflicts with a different version")), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<string[]>(),It.IsAny<MessageBoxImage>()))
+                .Returns(MessageBoxResult.Cancel);
+             dlgMock.Setup(m => m.Show(It.Is<string>(x => x.Contains("Are you sure you want to install")), It.IsAny<string>(), It.IsAny<MessageBoxButton>(), It.IsAny<MessageBoxImage>()))
+                .Returns(MessageBoxResult.OK);
+            
+            MessageBoxService.OverrideMessageBoxDuringTests(dlgMock.Object);
+
+            //
+            // 1. User downloads a diff version of a loaded package in error state
+            //
+            {
+                var bltinpackagesPkgVers = VersionUtilities.PartialParse(pkg.VersionName);
+                var newPkgVers = new Version(bltinpackagesPkgVers.Major + 1, bltinpackagesPkgVers.Minor, bltinpackagesPkgVers.Build).ToString();
 
-            [Test]
+                var id = "test-123";
+                var deps = new List<Dependency>() { new Dependency() { _id = id, name = pkg.Name } };
+                var depVers = new List<string>() { newPkgVers };
+
+                mockGreg.Setup(m => m.ExecuteAndDeserializeWithContent<PackageVersion>(It.IsAny<Request>()))
+                .Returns(new ResponseWithContentBody<PackageVersion>()
+                {
+                    content = new PackageVersion()
+                    {
+                        version = newPkgVers,
+                        engine_version = pkg.EngineVersion,
+                        name = pkg.Name,
+                        id = id,
+                        full_dependency_ids = deps,
```

Comment:
The name of the package being downloaded is the same as its dependency? How can it depend on itself?

Reply:
full_dependency_ids contain the package itself.